### PR TITLE
feat: add support for navigation as content

### PR DESCRIPTION
**Pull request by rs-amp**
Adds content managed navigation and related helper methods.

- Navigation Content Types
  - Content Page Link
  - Category Link
     - Points to category on SFCC. Can optionally populate the children of a category nav item from SFCC, and treat child content items as overrides.
  - Internal Link
  - External Link
  - Group
- Navigation content in automation.
  - `main-nav` for header navigation.
  - `footer-nav` for footer navigation.
- Navigation items can be hidden and reordered.
- Changes to navigation can be visualized in real time, on header and footer navs.
- Realtime Visualization context use simplified by new `useAmpRtv` method from utils/rtv.
- Documentation
- More misc refactoring, style changes and bugfixes.

### CODE REVIEW

File: app/components/amplience/hero/navigationHero.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            name: PropTypes.string,
+            endpoint: PropTypes.string,
+            defaultHost: PropTypes.string
+        }),
+        alt: PropTypes.string
+    }),
+    /**
+     * Hero component main title
+     */
+    // title: PropTypes.string,
```
Comment (rs-amp) [inline app/components/amplience/hero/navigationHero.tsx:83]: Comment unnecessary?

File: app/ssr.js [hunk 2]
```diff
@@ -80,7 +93,11 @@ const {handler} = runtime.createHandler(options, (app) => {
     app.get('/favicon.ico', runtime.serveStaticFile('static/ico/favicon.ico'))
 
     app.get('/worker.js(.map)?', runtime.serveServiceWorker)
-    app.get('*', runtime.render)
+
+    app.get('*', async (req, res) => {
+        console.log(`\n\n\nreq url: ${req.url}`)
```
Comment (rs-amp) [inline app/ssr.js:98]: This should probably be reverted.

File: config/amplience/default.js [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module.exports = {
-    hub: 'sfcccomposable'
+    hub: 'sfcccompdev'
```
Comment (rs-amp) [inline config/amplience/default.js:2]: Remove this before merge.

File: scripts/amplience/clean.ts [hunk 4]
```diff
@@ -99,8 +114,25 @@ export const cleanHandler = async (context: Arguments<Context>): Promise<any> =>
     }))
 
     console.log('Deleting mapFile...')
-    rmSync(mappingFile, {force: true})
+    rmSync(mappingFile, { force: true })
 
-    console.log(`Done!`)
+    // novadev-581 Update automation so that cleanup removes all folders from repositories
```
Comment (rs-amp) [inline scripts/amplience/clean.ts:119]: Weird to have this as a comment.

File: scripts/amplience/amplience-rest-client.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            if (error.response?.status === 429) {
+                await sleep(1000)
+                return await request(method)(config)
+            }
+            else if (error.response?.status === 404) {
+                // don't throw on a 404 just return an empty result set
+                return null
+            }
+
+            // if (error.stack) {
```
Comment (rs-amp) [inline scripts/amplience/amplience-rest-client.ts:72]: Remove?

File: scripts/amplience/amplience-rest-client.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+type AuthenticationStatus = 'NOT_LOGGED_IN' | 'LOGGING_IN' | 'LOGGED_IN'
+export const AmplienceRestClient = (config: { clientId: string, clientSecret: string }): OAuthRestClientInterface => {
+    let authenticatedAxios: AxiosInstance
+    let status: AuthenticationStatus = 'NOT_LOGGED_IN'
+
+    const apiUrl = `https://api.amplience.net/v2/content`
+    const authUrl = `https://auth.amplience.net/oauth/token?client_id=${config.clientId}&client_secret=${config.clientSecret}&grant_type=client_credentials`
+
+    const authenticate = async (): Promise<AxiosInstance> => {
+        // console.log(`authenticating to ${config.auth_url}`)
```
Comment (rs-amp) [inline scripts/amplience/amplience-rest-client.ts:20]: Remove?

**Comment by rs-amp** [comment]
The project generally is inconsistent with spaces/tabs in automation JSON. Should probably pick one and change all the files.

File: amplience-automation/automation-files/content/sitestructure/01 - Main Navigation/Our Jewelry.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
+{
+  "id": "73249be6-4dd4-43dc-b4f9-d3ffffe7340c",
+  "body": {
+    "_meta": {
+      "name": "Jewelry 2",
```
Comment (rs-amp) [inline amplience-automation/automation-files/content/sitestructure/01 - Main Navigation/Our Jewelry.json:5]: Maybe this should be changed to "Our Jewelry"?

File: app/components/amplience/hero/navigationHero.tsx (hunk 1 above)
Comment (rezakalfane) [reply app/components/amplience/hero/navigationHero.tsx:83]: yes. unnecessary :)

File: app/ssr.js (hunk 2 above)
Comment (rezakalfane) [reply app/ssr.js:98]: Yes, this was probably for debug from Dave

File: config/amplience/default.js (hunk 3 above)
Comment (rezakalfane) [reply config/amplience/default.js:2]: Yes

File: scripts/amplience/clean.ts (hunk 4 above)
Comment (rezakalfane) [reply scripts/amplience/clean.ts:119]: We can clean these comments as the branches are removed anyway

File: scripts/amplience/amplience-rest-client.ts (hunk 5 above)
Comment (rezakalfane) [reply scripts/amplience/amplience-rest-client.ts:72]: I'd say yes remove

File: scripts/amplience/amplience-rest-client.ts (hunk 6 above)
Comment (rezakalfane) [reply scripts/amplience/amplience-rest-client.ts:20]: yes

File: amplience-automation/automation-files/content/sitestructure/01 - Main Navigation/Our Jewelry.json (hunk 7 above)
Comment (rezakalfane) [reply amplience-automation/automation-files/content/sitestructure/01 - Main Navigation/Our Jewelry.json:5]: Yes

File: .gitignore [hunk 8]
```diff
@@ -33,3 +33,4 @@ app/loader-cache-hash-manifest.json
 
 .vscode
 .DS_Store
+.DS_Store
```
[the remark this answers is not in the document]
Comment (rezakalfane) [reply .gitignore:36]: should be only once :)

File: amplience-automation/automation-files/schema/schemas/navigation-content-page.json [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  "title": "Navigation Content Page",
+  "description": "Navigation item for content page",
+  "allOf": [
+    {
+      "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content"
+    }
+  ],
+  "trait:hierarchy": {
+    "childContentTypes": [
+			"https://sfcc.com/site/navigation/external",
```
[the remark this answers is not in the document]
Comment (rezakalfane) [reply amplience-automation/automation-files/schema/schemas/navigation-content-page.json:13]: We can replace all the tabs to double spaces in all files

**Review by rs-amp — COMMENTED** [review]
LGTM, can't approve as I made the request though.

**Pull request merged**